src/client/datascience/variablesView/variableViewActivationService.ts

```diff
@@ -18,9 +18,8 @@ export class VariableViewActivationService implements IExtensionSingleActivation
         // Only activate this when in the NativeNotebook experiment
         if (await this.experimentService.inExperiment(Experiments.NativeNotebook)) {
             this.extensionContext.subscriptions.push(
-                // Consider not using retainContext here? This will save the context of our view, but take more memory in VS Code.
                 window.registerWebviewViewProvider(this.variableViewProvider.viewType, this.variableViewProvider, {
-                    webviewOptions: { retainContextWhenHidden: true }
+                    webviewOptions: { retainContextWhenHidden: false }
```

Comment:
So the bug was a bit more interesting than I thought at first. Turns out I was actually keeping the view and execution count updated. But with retaincontext on the view was updating state when hidden. 

When the view was made visible after executing it refreshed, but after the refresh (since the state was already updating while the view was not visible) it didn't trigger a re-render since there wasn't a state change. You could actually see this if you moused over the view in this state it would re-render and have the correct values.

I already had a bug #4379 to consider not using retain context. This does mean that it might be a tiny bit slowing opening this view for a second / third / forth time, but it's not very noticeable as the variable fetch is the slow part, not the UI initialization.

Reply:
I think you should add a comment here saying why it has to be false. Somebody might change this later and not realize it messes up the execution count.

Reply:
@rchiodo Will do. Also I think that it's just the default (since vscode doesn't want views to retain context) so maybe I can comment and just remove the options object.